turtle_graphics/turtle_graphics.js

```diff
@@ -0,0 +1,310 @@
+        } else if (this.direction === 'SOUTH') {
+            this.direction = 'WEST';
+        }
+        return this;
+        //console.log('Current direction AFTER turning RIGHT: ' + this.direction);
+    }
+
+    left() {
+
+        //console.log('Current direction BEFORE turning LEFT: ' + this.direction);
+        if (this.direction === 'EAST') {
+            this.direction = 'NORTH';
+        } else if (this.direction === 'WEST') {
+            this.direction = 'SOUTH';
+        } else if (this.direction === 'NORTH') {
+            this.direction = 'WEST';
+        } else if (this.direction === 'SOUTH') {
+            this.direction = 'EAST';
+        }
+        return this;
+        //console.log('Current direction AFTER turning LEFT: ' + this.direction);
+
+    }
+
+    allPoints() {
+
+        //simply returns already existing allPositions array that belongs to the turtle class
+        
+        //let length = this.allPositions.length;
+        //console.log("Current allPosition length: " + length);
+        //if (length >= 0) { //i.e. the turtle has moved at all
+        //for (let i = 0; i < length; i++) {
+          // console.log(this.allPositions[i]);
+        //}
+        
+        return this.allPositions;
+
+    }
+
+    print() {
+
+        let allPoints = this.allPositions;
+
+        let maxX = Math.max(...allPoints.map(element => {
+            return element[0];
+        })); //columns
+
+        //console.log("Max x: " + maxX);
+
+        let maxY = Math.max(...allPoints.map(element => {
+            return element[1];
+        }));//rows
+
+        //console.log("Max y: " + maxY);
+
+        let a1 = []; //temporarily array of array to hold all x,y grid coordinates
+        
+        //build grid [row x column]
+        for(let x = 0; x <= maxY; x++){ //building rows
+            let a2 = [];    
+            for(let y = 0; y <= maxX; y++){ //building columns for each row
+                a2.push('.');
+            };
+            //console.log("temp a2: " + a2);
+            a1.push(a2);
+           
+        };
+        
+        //printing the grid for debugging
+        var iterator = a1.entries();
+        for (let i= 0; i < a1.length; i++) {
+            //console.log("a1: " + iterator.next().value);
+        }
+        
+        //allPoints.forEach((point) => {
+          //  console.log(a1[point[0]][point[1]]);
+          //  console.log(a1[point[1]][point[0]] = 'x');
+        //});
+
+         for (let i = 0; i < allPoints.length; i++) {
+            //console.log("AllPoint element : " + allPoints[i]);
+            //console.log("AllPoint element Column value: " + allPoints[i][0]);
+            let x = allPoints[i][0];
+            //console.log("AllPoint element ROW value: " + allPoints[i][1]);
+            let y = allPoints[i][1];
+            // must pass row x column to update the value; that is why y and x are reversed
+            //console.log(a1[y][x] = '@');
+            
+            a1[y][x] = '@'
```

Comment:
Does not handle "out of bounds" coordinates. This isn't an issue with the assignment, but as you progress in your career it will be important to consider which errors are likely to occur and how to handle them.